### PR TITLE
FATES API 22

### PR DESCRIPTION
**Pull request by rgknox**
This set of changes brings E3SM into compatability with FATES API 21, and updates the pointer to tag: sci.1.53.0_api.21.0.0.  Recent API changes accomodate an overhaul to FATES history variables (including new testdef definitions on E3SM side), passing of canopy snow fraction to FATES, and accomodation of history reporting of variable LAI bin widths in FATES.

Fixes Issue #4755 and Fixes Issue #4816

[non-BFB] for FATES tests

### CODE REVIEW

**Comment by rgknox** [comment]
Tested e3sm_land_developer regression tests on cori, against E3SM hash 5834afad02. All non-FATES should be B4B, FATES tests should all have different results, including different NLCOMPS.  Results looked good.

**Comment by peterdschwartz** [comment]
Does this update include the fix for Issue #4755 (the satphen failure with GNU)?

**Comment by glemieux** [comment]
@peterdschwartz I've got FATES PR https://github.com/NGEET/fates/pull/838 up to fix this.  It's isolated to the fates side only.  We should probably be able to get this integrated for fates api21.

**Comment by peterdschwartz** [comment]
@glemieux Great!  Based on your comment in the Fates repo, will this be nonBFB for all compiler satphen tests?

**Comment by rgknox** [comment]
Note, some work needs to be done on this PR before this branch is compatible with that new FATES tag.  I will let you know @peterdschwartz when its ready

**Comment by peterdschwartz** [comment]
@rgknox  thanks for clarifying.   I suppose I can also be a reviewer from the ELM side.  Would you mind requesting someone suitable to review as well?

**Comment by rgknox** [comment]
@peterdschwartz, since we shouldn't be impacting any non-fates tests, probably @glemieux is the best reviewer.  Did you want an E3SM-side reviewer though?

**Comment by rgknox** [comment]
@peterdschwartz this branch has all the intended content now, feel free to proceed.  I ran the land developer test suite on cori and things looked as expected, no FATES tests should be b4b, all other tests should be b4b.

/global/cscratch1/sd/rgknox/e3sm_tests/E1efd6f7704-Ffccbd303-fates-api22-v3

**Review by peterdschwartz — APPROVED** [review]

File: components/elm/src/main/elmfates_interfaceMod.F90 [hunk 1]
```diff
@@ -827,8 +842,6 @@ subroutine dynamics_driv(this, bounds_clump, top_as_inst,          &
 
          do ifp = 1, this%fates(nc)%sites(s)%youngest_patch%patchno
             p = ifp+col_pp%pfti(c)
-            this%fates(nc)%bc_in(s)%t_veg24_pa(ifp) = &
```
Comment (glemieux) [inline components/elm/src/main/elmfates_interfaceMod.F90:830]: Why is it that we remove this?

**Review by glemieux — COMMENTED** [review]
One minor question in the comments.  Otherwise looks good.

File: components/elm/src/main/elmfates_interfaceMod.F90 (hunk 1 above)
Comment (ckoven) [reply components/elm/src/main/elmfates_interfaceMod.F90:830]: this was made redundant by https://github.com/NGEET/fates/pull/724

Comment (rgknox) [reply components/elm/src/main/elmfates_interfaceMod.F90:830]: yup, we don't even have t_veg24_pa anymore:  https://github.com/NGEET/fates/blob/master/main/FatesInterfaceTypesMod.F90

**Review by glemieux — APPROVED** [review]

**Comment by peterdschwartz** [comment]
Merged to master

**Comment by rgknox** [comment]
thanks @peterdschwartz

**Pull request merged**